src/irs/adapters/inbound/s3.py

```diff
@@ -47,87 +48,104 @@ def calc_part_ranges(
     return part_ranges
 
 
-async def get_download_url(*, object_id: str, bucket_id: str) -> str:
-    """Get object download URL from S3 inbox bucket"""
-    storage = get_objectstorage()
-    return await storage.get_object_download_url(
-        bucket_id=bucket_id, object_id=object_id
-    )
+@dataclass
+class S3IDs:
+    """Container for bucket and object ID"""
 
+    bucket_id: str
+    object_id: str
 
-async def get_object_size(*, object_id: str, bucket_id: str) -> int:
-    """Get object size from S3 metadata"""
-    storage = get_objectstorage()
-    return await storage.get_object_size(bucket_id=bucket_id, object_id=object_id)
 
+class StagingHandlerPort(ABC):
```

Comment:
Ports should go into the `ports` subpackage but, other than that, this setup nicely aligns with our 3hex pattern.